apps/backend/src/datasources/postgres/TemplateDataSource.ts

```diff
@@ -720,6 +720,15 @@ export default class PostgresTemplateDataSource implements TemplateDataSource {
     question: string,
     default_config: string
   ): Promise<Question> {
+    const naturalKeyExistAlready = await database
+      .select('natural_key')
+      .from('questions')
+      .where('natural_key', natural_key);
+
+    if (naturalKeyExistAlready.length != 0) {
+      natural_key = natural_key.concat('_' + Math.floor(Math.random() * 1000));
+    }
+

```

Comment:
I think it'd be a little more secure to use timestamp to avoid duplicated natural_key, but otherwise looks good to me

Reply:
Tried using it. But, the Date.now() is being taken as a function in concat and not appending the timestamp value. Value getting appended was like '_ now() ->..'. I believe the gap 1-1000 would be safe enough in this case.

Reply:
You can try using template literals the function maybe evaluated in case its the preference.

Reply:
Changed the expression to use timestamp.